plutarch-extra/Plutarch/Extra/Api.hs

```diff
@@ -0,0 +1,107 @@
+module Plutarch.Extra.Api (
+  pgetContinuingOutputs,
+  pfindOwnInput,
+  pfindDatum,
+) where
+
+import Plutarch.Api.V1 (
+  PAddress,
+  PDatum,
+  PDatumHash,
+  PTuple,
+  PTxInInfo,
+  PTxOut,
+  PTxOutRef,
+ )
+import Plutarch.Prelude
+
+{- | Find the output txns corresponding to the input being validated.
+
+  Takes as arguments the inputs, outputs and the spending transaction referenced
+  from `PScriptPurpose`.
+
+  __Example:__
+
+  @
+  ctx <- tcont $ pletFields @["txInfo", "purpose"] sc
+  pmatchC (getField @"purpose" ctx) >>= \case
+    PSpending outRef' -> do
+      let outRef = pfield @"_0" # outRef'
+          inputs = pfield @"inputs" # (getField @"txInfo" ctx)
+          outputs = pfield @"outputs" # (getField @"txInfo" ctx)
+      pure $ pgetContinuingOutputs # inputs # outputs # outRef
+    _ ->
+      pure $ ptraceError "not a spending tx"
+  @
+-}
+pgetContinuingOutputs :: Term s (PBuiltinList (PAsData PTxInInfo) :--> PBuiltinList (PAsData PTxOut) :--> PTxOutRef :--> PBuiltinList PTxOut)
+pgetContinuingOutputs = phoistAcyclic $
+  plam $ \inputs outputs outRef ->
+    pmatch (pfindOwnInput # inputs # outRef) $ \case
+      PJust tx -> do
+        let resolved = pfield @"resolved" # tx
+            outAddr = pfield @"address" # resolved
+        pfilter # (matches # outAddr) #$ pmap # plam pfromData # outputs
```

Comment:
this should be a singular `pfilter` yeah? that `pmap` is morally a no-op.

Reply:
@TotallyNotChase How would that be a no-op? Where else would you `pfromData` the elements of `outputs`?

Or do you mean making this function return ` PBuiltinList (PAsData PTxOut)`?

Reply:
> Or do you mean making this function return PBuiltinList (PAsData PTxOut)?
yep!